### ISSUE TITLE
Several issues during training on my own dataset

### ISSUE THREAD

**Issue by xuleiyang** [yhenon/keras-frcnn#24]
Hi yhenon, 

I got two issues during training on my own dataset, would appreciated if you could advise here.

1) My workstation got 4 GPU cards, it seems one of them is got used, so my question is how do I configure to make full use of 4 GPU cards (I am running tensorflow backend)

2) Is it ok if I did not do any annotation for background (i.e., background example number is zero)? I meet some exception during the training as follows, but it did not stop the training.  Do you have any idea what could lead to this kind of exception. 
Exception: 'NoneType' object has no attribute '__getitem__'

Many Thanks!

Xulei

**Comment by xuleiyang** [comment]
3. which index during the training process is indicating whether the training progress is good or not?

Thank you!

**Comment by yhenon** [comment]
1 - multi GPU support is coming soon. The data generation is quite slow right now, so some improvements need to be made on that side.
2 - no bg annotation is ok, in fact it's the most common case. You only add bg annotations if you are interested in hard negative mining.
3 - all the losses should ideally go down. faster-rcnn will really only work if both the RPN and classifier have converge satisfactorily. Here's what my output looks like when it's trained well:
```
Mean number of bounding boxes from RPN overlapping ground truth boxes: 22.4260528893
Classifier accuracy for bounding boxes from RPN: 0.8285
Loss RPN classifier: 0.0820875049883
Loss RPN regression: 0.0902734231516
Loss Classifier classifier: 0.435039975807
Loss Classifier regression: 0.160380041832
```

**Comment by ahmedbaracat** [comment]
@yhenon about the no bg annotation... what if I am training using images that don't have any annotation (because they don't contain any of the classes I am interested in), `simple_parser.py` will not load them because they aren't included in the annotations file. How to go about making sure that these images are included in the training?

**Comment by yhenon** [comment]
Images with no class annotations do get ignored currently (which matches
the original implementation).
I think this would be a good feature to add.
You could either submit a pull request, or open an issue and I'll add it
soon

**Comment by anmolkalia** [comment]
Hey @yhenon  ! I am getting a lot of 'NoneType' object has no attribute '__getitem__' exceptions and I cannot figure out why. Is this an issue? What could be the reason behind this? I am using simple_parser to get the data.

![screenshot from 2017-04-30 11-29-12](https://cloud.githubusercontent.com/assets/7467123/25565611/50230a1a-2d98-11e7-9bd5-d7f4f7143571.png)

**Comment by Poojadj** [comment]
Code is implemented in python2, but you are using python3 version i guess so. You can change the code from py2 to py3

**Comment by EstellaD** [comment]
@yhenon @ahmedbaracat Hi, so can I train my own data set without any annotation file now? My training pictures are single items, so I don't think it's necessary to draw a bounding box around it.

**Comment by DanMossa** [comment]
@EstellaD  if the training images are single files maybe you could just kinda create a text file with 

`filename,0,0,width,height,label`
for all the images

**Comment by EstellaD** [comment]
Hi Daniel, thank you so much for the help! Estella

On Fri, Jul 21, 2017 at 10:25 PM, Daniel Mossaband <notifications@github.com
